packages/backend/src/core/activitypub/models/ApPersonService.ts

```diff
@@ -356,10 +359,13 @@ export class ApPersonService implements OnModuleInit {
 				}));
 
 				if (person.publicKey) {
+					// TODO: 一人のユーザが複数の公開鍵を持っている場合があるので、MiUserPublicKeyの主キーもuserIdから(userId, keyId)に変更する必要があるかも…？
+					// As a user can have multiple public keys, it might be better to change MiUserPublicKey's primary key from userId to (userId, keyId).
+					const publicKey = Array.isArray(person.publicKey) ? person.publicKey[0] : person.publicKey;
```

Comment:
この処理だと配列が空の場合に`publicKey`が`undefined`になってしまうので、配列からの値の取り出しは`if`の手前で行うのが確実です。（そもそも公開鍵を持たないのに`publicKey`を指定しているactorという時点で奇妙なので、エッジケースではありますが。）

ついでに前のレビューコメントと同様に`misc`の関数を使った方が良いかと思います（要`import { toSingle } from '@/misc/prelude/array.js';`）。

```suggestion
				// TODO: 一人のユーザが複数の公開鍵を持っている場合があるので、MiUserPublicKeyの主キーもuserIdから(userId, keyId)に変更する必要があるかも…？
				// As a user can have multiple public keys, it might be better to change MiUserPublicKey's primary key from userId to (userId, keyId).
				const publicKey = toSingle(person.publicKey);
				if (publicKey) {
```